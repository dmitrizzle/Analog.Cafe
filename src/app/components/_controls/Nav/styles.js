// tools
import React from "react"
import styled from "styled-components"

// components
import Link from "../Link"

// NOTE: *most* CSS has bee copied into /public/index.html
// for faster critical path loading times.
// some actionable CSS remains here.

// css

// const containerVerticalSpacing = props => props.theme.size.font.make.larger / 1.2

export const NavWrapper = styled.nav`
  ${"" /* ${props => props.theme.size.font.auto} ${props =>
      props.theme.typography.title.auto} position: relative;
  z-index: ${props => props.theme.layer.nav};
  margin-top: ${containerVerticalSpacing}em;
  margin-bottom: ${containerVerticalSpacing}em;

  > ul {
    padding: 0;
    display: flex;
    align-items: center;
    justify-content: center;
    margin: 0 auto;
  } */};
`
const StyledLink = styled(Link)`
  ${"" /* background: 			${props => props.theme.color.background};
	text-decoration: 	none;
	position: 				relative; */};
  &.active::before {
    content: "";
    width: 110%;
    left: -5%;
    height: 2px;
    bottom: -5px;
    background: ${props => props.theme.color.foreground};
    position: absolute;
  }
  ${props =>
    props.connectionStatus === "offline"
      ? `
    opacity: ${props.theme.opacity.half}
  `
      : null};
`
export const NavLink = props => {
  return <StyledLink {...props} activeClassName="active" />
}
export const NavIndexLink = props => {
  return <Link {...props} activeClassName="active" />
}

export const NavItem = styled.li`
  ${"" /* display: block;
  text-align: center;
  width: 7em; */} ${"" /* ${props =>
    props.center
      ? props => props.theme.size.breakpoint.max.m`order: 1;`
      : false} ${props =>
      props.left
        ? props => props.theme.size.breakpoint.max.m`order: 0;`
        : false} ${props =>
      props.right
        ? props => props.theme.size.breakpoint.max.m`order: 2;`
        : false} ${props =>
      props.prime
        ? false
        : props => props.theme.size.breakpoint.max.m`display:none;`} ${props =>
      props.indicator
        ? `color: ` +
          props.theme.color.alpha.foreground(props.theme.opacity.half)
        : null}; */};

  ${props =>
    props.draftStatus &&
    `
    // for "Draft Saved" button we need to set some trims because on very small
    // screens it breaks design
    overflow: hidden;
    height: ${props.theme.size.block.spacing /
      props.theme.size.font.make.smaller}em;
  `};
`

// styles for tiny profile icon
export const TinyImageInline = styled.span`
  width: 1em;
  height: 1em;
  border-radius: 1em;
  margin: 0 0 -0.075em;
  overflow: hidden;
  display: inline-block;

  background: ${props =>
    props.theme.color.alpha.foreground(props.theme.opacity.least)};
  box-shadow: 0 0 0 1px
    ${props => props.theme.color.alpha.foreground(props.theme.opacity.half / 2)}
    inset;

  & > img {
    width: 100%;
    display: block;
  }
`

// styles for connectivity indicator
export const Connection = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  padding: ${props => props.theme.size.block.spacing / 4}em
    ${props => props.theme.size.block.spacing / 2}em
    ${props => props.theme.size.block.spacing / 3}em
    ${props => props.theme.size.block.column.safety / 2}em;
  ${props => props.theme.typography.title.auto} font-size: ${props =>
      props.theme.size.font.make.smaller}em;
  background: ${props => props.theme.color.brand};
  color: ${props => props.theme.color.background};
  border-bottom-left-radius: ${props => props.theme.size.block.spacing / 2}em;
`
