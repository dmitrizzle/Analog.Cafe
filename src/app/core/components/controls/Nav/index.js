import { connect } from "react-redux"
import React from "react"

import { setUserIntent } from "../../../../user/store/actions-user"
import NavComposer from "./components/NavComposer"
import NavConnectionStatus from "./components/NavConnectionStatus"
import NavGeneral from "./components/NavGeneral"
import NavWrapper from "./components/NavWrapper"

const Nav = props => {
  if (props.top && !props.nav.location.top) return null
  if (props.bottom && !props.nav.location.bottom) return null

  if (props.top && props.nav.location.top)
    return [
      <NavConnectionStatus
        key="Online_status"
        style={{
          display: props.user.connection.status === "offline" ? "block" : "none"
        }}
      >
        Offline
      </NavConnectionStatus>,
      <NavWrapper className="appNav" key="NavWrapper">
        {props.nav.view === "COMPOSER" ? (
          <NavComposer
            composerStatus={props.composer.status}
            editorStatus={props.editor.status}
            userStatus={props.user.status}
            userRole={props.user.info.role}
          />
        ) : (
          <NavGeneral
            userStatus={props.user.status}
            userRole={props.user.info.role}
            userImage={props.user.info.image}
            userIntent={() => {
              props.user.intent.load !== "List" &&
                props.setUserIntent({ load: "List" })
            }}
            articleTag={props.article.tag}
            modalUrl={props.modal.requested.url}
            isModalHidden={props.modal.hidden}
          />
        )}
      </NavWrapper>
    ]
  if (props.bottom && props.nav.location.bottom) return null // no footer
}

const mapStateToProps = state => {
  return {
    nav: state.nav,
    composer: state.composer,
    editor: state.editor,
    user: state.user,
    article: state.article,
    modal: state.modal
  }
}
const mapDispatchToProps = dispatch => {
  return {
    setUserIntent: intent => {
      dispatch(setUserIntent(intent))
    }
  }
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Nav)
