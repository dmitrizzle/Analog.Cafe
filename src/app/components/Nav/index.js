// tools
import React from "react"

// components
import Logo from "../Logo"
import done from "../_icons/images/done.jpg"
import { ModalDispatch } from "../../containers/Modal"

import { ROUTE_AUTH_USER_LANDING } from "../../../constants/user"

// styles
import { NavLink, NavIndexLink, NavItem } from "./styles"

// return
export const CommonNav = props => {
  return (
    <ul>
      <NavItem>
        <NavLink to={"/photo-essays"}>
          <span>Photo Essays</span>
        </NavLink>
      </NavItem>
      <NavItem>
        <NavLink to={"/articles"}>
          <span>Articles</span>
        </NavLink>
      </NavItem>
      <NavItem prime center>
        <NavIndexLink to={"/"}>
          <Logo />
        </NavIndexLink>
      </NavItem>
      <NavItem prime left>
        {props.userStatus === "ok" ? (
          <NavLink to={"/submit/compose"}>
            <span>Submit</span>
          </NavLink>
        ) : (
          <NavLink to={"/about"}>
            <span>About</span>
          </NavLink>
        )}
      </NavItem>
      <NavItem prime right>
        {props.userStatus === "ok" ? (
          <NavLink to={ROUTE_AUTH_USER_LANDING}>
            <span>My Stuff</span>
          </NavLink>
        ) : (
          <NavLink to={"/submit"}>
            <span>Submit</span>
          </NavLink>
        )}
      </NavItem>
    </ul>
  )
}

const NavLinkSendLabel = () => {
  return <span>Send</span>
}
const NavLinkSend = props => {
  return (
    <ModalDispatch
      with={{
        request: {
          url: "/api/messages/submit-consent.json"
        }
      }}
      style={{ textDecoration: "none" }}
    >
      <NavLinkSendLabel />
    </ModalDispatch>
  )
}

export const ComposerNav = props => {
  return (
    <ul>
      <NavItem indicator prime left>
        <ModalDispatch
          with={{
            info: {
              image: done,
              title: "Never Loose Your Work!",
              text:
                "Your draft is saved automatically every 3 seconds onto your device. Even when you arenâ€™t connected to the internet."
            },
            id: "hints/save"
          }}
          style={{ textDecoration: "none" }}
        >
          <span>{props.draftStatus}</span>
        </ModalDispatch>
      </NavItem>
      <NavItem prime center>
        <NavIndexLink to={"/"}>
          <Logo />
        </NavIndexLink>
      </NavItem>
      <NavItem prime right>
        <NavLinkSend userStatus={props.userStatus} />
      </NavItem>
    </ul>
  )
}

export { NavWrapper } from "./styles"
