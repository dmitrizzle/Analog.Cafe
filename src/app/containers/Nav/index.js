// tools
import React from "react"

// redux
import { connect } from "react-redux"
import { setIntent as setUserIntent } from "../../../actions/userActions"

// components
import { CommonNav, ComposerNav, NavWrapper } from "../../components/Nav"

// render
const Nav = props => {
  if (props.top && !props.nav.location.top) return null
  if (props.bottom && !props.nav.location.bottom) return null
  return (
    <NavWrapper className="appNav">
      {props.nav.view === "COMPOSER" ? (
        <ComposerNav
          draftStatus={props.composer.draftStatus}
          userStatus={props.user.status}
        />
      ) : (
        <CommonNav
          userStatus={props.user.status}
          userImage={props.user.info.image}
          userIntent={() => {
            props.user.intent.load !== "List" &&
              props.setUserIntent({ load: "List" })
          }}
        />
      )}
    </NavWrapper>
  )
}

// connect with redux
const mapStateToProps = state => {
  return {
    nav: state.nav,
    composer: state.composer,
    user: state.user
  }
}
const mapDispatchToProps = dispatch => {
  return {
    setUserIntent: intent => {
      dispatch(setUserIntent(intent))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Nav)
