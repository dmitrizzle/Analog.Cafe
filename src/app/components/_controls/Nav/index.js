// tools
import React from "react"
import { froth } from "../../../../utils/image-froth"

// components
import Logo from "../../_icons/Logo"
import { ModalDispatch } from "../../../containers/Modal"

import {
  MESSAGE_HINT_SUBMIT_CONSENT,
  MESSAGE_HINT_AUTO_SAVE
} from "../../../../constants/messages/hints"

// styles
import { NavLink, NavIndexLink, NavItem, TinyImageInline } from "./styles"

// NOTE: components' `className` props are used in index.html and aren't required
// if styles from `./styles.js` have the comments removed

// return

// tiny profile image icon for logged-in users
const TinyImage = props => {
  return (
    <TinyImageInline>
      <img
        src={froth({ src: props.image, size: "i" }).src}
        alt="Profile avatar"
      />
    </TinyImageInline>
  )
}

// Nav components
export const CommonNav = props => {
  return (
    <ul
      onMouseOver={
        "ontouchstart" in document.documentElement ? null : props.userIntent
      }
    >
      <NavItem>
        <NavLink to={"/photo-essays"}>
          <span>Photo Essays</span>
        </NavLink>
      </NavItem>
      <NavItem>
        <NavLink to={"/articles"}>
          <span>Articles</span>
        </NavLink>
      </NavItem>
      <NavItem prime center className="prime center">
        <NavIndexLink to={"/"} className="indexRouteLink">
          <Logo />
        </NavIndexLink>
      </NavItem>
      <NavItem prime right className="prime left">
        {props.userStatus === "ok" ? (
          <NavLink
            to={
              process.env.NODE_ENV === "development"
                ? "/submit/compose"
                : "/beta/compose"
            }
          >
            <span>Submit</span>
          </NavLink>
        ) : (
          <NavLink to={"/submit"}>
            <span>Submit</span>
          </NavLink>
        )}
      </NavItem>
      <NavItem prime left className="prime right">
        <ModalDispatch
          with={{
            info: {
              title: "More…",
              buttons: [
                props.userStatus === "ok"
                  ? {
                      to: "/me",
                      text: "My Submissions",
                      red: true
                    }
                  : null,
                props.userStatus === "ok"
                  ? {
                      to: "/me/edit",
                      text: "My Profile",
                      red: true
                    }
                  : null,
                props.userStatus === "ok"
                  ? {
                      to: "/sign-out",
                      text: "Sign Out",
                      black: true
                    }
                  : null,
                {
                  to: "/photo-essays",
                  text: "Photo Essays",
                  responsiveMobileOnly: true
                },
                {
                  to: "/editorials",
                  text: "Editorials"
                },
                {
                  to: "/guides",
                  text: "Guides"
                },
                {
                  to: "/reviews",
                  text: "Reviews"
                },
                {
                  to: "/stories",
                  text: "Stories"
                },
                { divider: true },
                {
                  to: "/collaborations",
                  text: "Collaborations"
                },
                {
                  to: "/solo-projects",
                  text: "Solo Projects"
                },
                { divider: true },
                {
                  to: "/about",
                  text: "About Analog.Cafe"
                }
              ]
            },
            id: "nav/more"
          }}
          style={{ textDecoration: "none", paddingRight: ".25em" }}
        >
          <span>
            More…{" "}
            {props.userStatus === "ok" && <TinyImage image={props.userImage} />}
          </span>
        </ModalDispatch>
      </NavItem>
    </ul>
  )
}

const NavLinkSendLabel = () => {
  return <span>Send</span>
}
const NavLinkSend = props => {
  return (
    <ModalDispatch
      with={MESSAGE_HINT_SUBMIT_CONSENT}
      style={{ textDecoration: "none" }}
    >
      <NavLinkSendLabel />
    </ModalDispatch>
  )
}

export const ComposerNav = props => {
  return (
    <ul>
      <NavItem indicator prime left className="prime left">
        <ModalDispatch
          with={MESSAGE_HINT_AUTO_SAVE}
          style={{ textDecoration: "none" }}
        >
          <span>{props.draftStatus}</span>
        </ModalDispatch>
      </NavItem>
      <NavItem prime center className="prime center">
        <NavIndexLink to={"/"} className="indexRouteLink">
          <Logo />
        </NavIndexLink>
      </NavItem>
      <NavItem prime right className="prime right">
        <NavLinkSend userStatus={props.userStatus} />
      </NavItem>
    </ul>
  )
}

export { NavWrapper } from "./styles"
