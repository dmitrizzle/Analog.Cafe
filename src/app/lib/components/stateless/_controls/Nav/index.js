// tools
import React from "react"
import { froth } from "../../../../utils/image-froth"

// components
import Logo from "../../_icons/Logo"
import { ModalDispatch } from "../../../containers/Modal"

import {
  MESSAGE_HINT_SUBMIT_CONSENT,
  MESSAGE_HINT_SUBMIT_EDITORS,
  MESSAGE_HINT_AUTO_SAVE
} from "../../../../constants/messages/hints"

// styles
import { NavLink, NavIndexLink, NavItem, TinyImageInline } from "./styles"

// NOTE: components' `className` props are used in index.html and aren't required
// if styles from `./styles.js` have the comments removed

// return

// tiny profile image icon for logged-in users
const TinyImage = props => {
  return (
    <TinyImageInline>
      <img
        src={froth({ src: props.image, size: "i" }).src}
        alt="Profile avatar"
      />
    </TinyImageInline>
  )
}

// Nav components
const NavLinkLabelMore = props => {
  return (
    <span>
      More… {props.userStatus === "ok" && <TinyImage image={props.userImage} />}
    </span>
  )
}
export const CommonNav = props => {
  return (
    <ul
      onMouseOver={
        "ontouchstart" in document.documentElement ? null : props.userIntent
      }
    >
      <NavItem>
        <NavLink to={"/photo-essays"}>
          <span>Photo Essays</span>
        </NavLink>
      </NavItem>
      <NavItem>
        <NavLink to={"/articles"}>
          <span>Articles</span>
        </NavLink>
      </NavItem>
      <NavItem prime center className="prime center">
        <NavIndexLink to={"/"} className="indexRouteLink">
          <Logo />
        </NavIndexLink>
      </NavItem>
      <NavItem prime right className="prime left">
        {props.userStatus === "ok" ? (
          <NavLink to={"/submit/compose"}>
            <span>Submit</span>
          </NavLink>
        ) : (
          <NavLink to={"/submit"}>
            <span>Submit</span>
          </NavLink>
        )}
      </NavItem>
      <NavItem prime left className="prime right">
        <ModalDispatch
          with={{
            info: {
              title: "More…",
              subscribeForm: props.userStatus !== "ok",
              buttons: [
                props.userStatus === "ok"
                  ? {
                      to: "/me",
                      text: "My Submissions",
                      branded: true
                    }
                  : null,
                props.userStatus === "ok"
                  ? {
                      to: "/me/edit",
                      text: "My Profile",
                      branded: true
                    }
                  : null,
                props.userRole === "admin"
                  ? {
                      to: "/me/admin",
                      text: "Admin",
                      branded: true
                    }
                  : null,
                props.userStatus === "ok"
                  ? {
                      to: "/sign-out",
                      text: "Sign Out",
                      inverse: true
                    }
                  : null,
                {
                  to: "/photo-essays",
                  text: "Photo Essays",
                  responsiveMobileOnly: true
                },
                {
                  to: "/editorials",
                  text: "Editorials"
                },
                {
                  to: "/guides",
                  text: "Guides"
                },
                {
                  to: "/reviews",
                  text: "Reviews"
                },
                {
                  to: "/stories",
                  text: "Stories"
                },
                { divider: true },
                {
                  to: "/collaborations",
                  text: "Collaborations"
                },
                {
                  to: "/solo-projects",
                  text: "Solo Projects"
                },
                { divider: true },
                {
                  to: "/about",
                  text: "About Analog.Cafe"
                }
              ]
            },
            id: "nav/more"
          }}
          style={{ textDecoration: "none", paddingRight: ".25em" }}
        >
          <NavLinkLabelMore
            userStatus={props.userStatus}
            userImage={props.userImage}
          />
        </ModalDispatch>
      </NavItem>
    </ul>
  )
}

const NavLinkSendLabel = () => {
  return <span>Send</span>
}
const NavLinkSend = props => {
  return (
    <ModalDispatch
      with={
        props.submissionStatus.id && props.userRole === "admin"
          ? MESSAGE_HINT_SUBMIT_EDITORS
          : MESSAGE_HINT_SUBMIT_CONSENT
      }
      style={{ textDecoration: "none" }}
    >
      <NavLinkSendLabel />
    </ModalDispatch>
  )
}

export const ComposerNav = props => {
  return (
    <ul>
      <NavItem draftStatus prime left className="prime left">
        <ModalDispatch
          with={MESSAGE_HINT_AUTO_SAVE}
          style={{ textDecoration: "none" }}
        >
          {props.draftStatus === "ok" && <span>Draft Saved</span>}
          {props.draftStatus === "pending" && <span>Saving…</span>}
          {!props.draftStatus && <span>Draft</span>}
        </ModalDispatch>
      </NavItem>
      <NavItem prime center className="prime center">
        <NavIndexLink to={"/"} className="indexRouteLink">
          <Logo />
        </NavIndexLink>
      </NavItem>
      <NavItem prime right className="prime right">
        <NavLinkSend
          userStatus={props.userStatus}
          userRole={props.userRole}
          submissionStatus={props.submissionStatus}
        />
      </NavItem>
    </ul>
  )
}

export { NavWrapper } from "./styles"
export { Connection } from "./styles"
