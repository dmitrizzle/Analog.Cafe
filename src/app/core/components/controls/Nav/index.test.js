import { createMockStore } from "redux-test-utils"
import { shallowWithStore } from "enzyme-redux"
import React from "react"

import Nav from "./"

it("Render Nav without crashing", () => {
  const store = createMockStore()
  shallowWithStore(<Nav />, store)
})
