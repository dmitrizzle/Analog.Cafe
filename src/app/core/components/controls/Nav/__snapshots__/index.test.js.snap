// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Nav, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Nav) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "article": undefined,
      "composer": undefined,
      "editor": undefined,
      "modal": undefined,
      "nav": undefined,
      "setUserIntent": [Function],
      "user": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "article": undefined,
        "composer": undefined,
        "editor": undefined,
        "modal": undefined,
        "nav": undefined,
        "setUserIntent": [Function],
        "user": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
